dspy/adapters/chat_adapter.py

```diff
@@ -54,40 +40,127 @@ def __call__(
             # fallback to JSONAdapter
             return JSONAdapter()(lm, lm_kwargs, signature, demos, inputs)
 
-    def format(
-        self, signature: Type[Signature], demos: list[dict[str, Any]], inputs: dict[str, Any]
-    ) -> list[dict[str, Any]]:
-        messages: list[dict[str, Any]] = []
-
-        # Extract demos where some of the output_fields are not filled in.
-        incomplete_demos = [
-            demo for demo in demos if not all(k in demo and demo[k] is not None for k in signature.fields)
-        ]
-        complete_demos = [demo for demo in demos if demo not in incomplete_demos]
-        # Filter out demos that don't have at least one input and one output field.
-        incomplete_demos = [
-            demo
-            for demo in incomplete_demos
-            if any(k in demo for k in signature.input_fields) and any(k in demo for k in signature.output_fields)
-        ]
-
-        demos = incomplete_demos + complete_demos
-        prepared_instructions = prepare_instructions(signature)
-        messages.append({"role": "system", "content": prepared_instructions})
-
-        # Add the few-shot examples
-        for demo in demos:
-            messages.append(self.format_turn(signature, demo, role="user", incomplete=demo in incomplete_demos))
-            messages.append(self.format_turn(signature, demo, role="assistant", incomplete=demo in incomplete_demos))
-
-        # Add the chat history after few-shot examples
-        if any(field.annotation == History for field in signature.input_fields.values()):
-            messages.extend(self.format_conversation_history(signature, inputs))
-        else:
-            messages.append(self.format_turn(signature, inputs, role="user"))
-
-        messages = try_expand_image_tags(messages)
-        return messages
+    def format_field_structure(self, signature: Type[Signature]) -> str:
+        """
+        `ChatAdapter` requires input and output fields to be in their own sections, with section header using markers
+        `[[ ## field_name ## ]]`.
+
+        For example:
+        ```
+        import dspy
+
+        class MySignature(dspy.Signature):
+            text: str = dspy.InputField(description="The text to analyze")
+            context: str = dspy.InputField(description="The context of the text")
```

Comment:
Why this long docstring that will go stale as soon as someone changes the adapter slightly?

Reply:
simplified for easy maintenance!